pkg/collect/longhorn.go

```diff
@@ -65,7 +65,8 @@ func (c *CollectLonghorn) Collect(progressChan chan<- interface{}) (CollectorRes
 	// collect nodes.longhorn.io
 	nodes, err := client.Nodes(ns).List(ctx, metav1.ListOptions{})
 	if err != nil {
-		return nil, errors.Wrap(err, "list nodes.longhorn.io")
+		logger.Printf("list nodes.longhorn.io not found")
```

Comment:
Not all errors will be "pod not found". There are many other [API errors](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L47) and runtime errors that can occur.

You want something like this

```go
	if err != nil {
		if apiErr, ok := err.(*apiErrors.StatusError); ok {
			if apiErr.ErrStatus.Code == 404 {
				logger.Printf("list nodes.longhorn.io not found")
				return nil, nil
			}
		}
		return nil, errors.Wrap(err, "list nodes.longhorn.io")
	}
```

Add this import as well
```go
	apiErrors "k8s.io/apimachinery/pkg/api/errors"
```

Reply:
updated accordingly